public/app/features/alerting/unified/hooks/useCombinedRule.ts

```diff
@@ -0,0 +1,115 @@
+import { AsyncRequestMapSlice, AsyncRequestState, initialAsyncRequestState } from '../utils/redux';
+import { useCombinedRuleNamespaces } from './useCombinedRuleNamespaces';
+import { useUnifiedAlertingSelector } from './useUnifiedAlertingSelector';
+import { fetchPromRulesAction, fetchRulerRulesAction } from '../state/actions';
+import { RulerRulesConfigDTO } from 'app/types/unified-alerting-dto';
+import * as ruleId from '../utils/rule-id';
+
+export function useCombinedRule(
+  identifier: RuleIdentifier | undefined,
+  ruleSourceName: string | undefined
+): AsyncRequestState<CombinedRule> {
+  const requestState = useCombinedRulesLoader(ruleSourceName);
+  const combinedRules = useCombinedRuleNamespaces(ruleSourceName);
+
+  const rule = useMemo(() => {
+    if (!identifier || !ruleSourceName || combinedRules.length === 0) {
+      return;
+    }
+
+    for (const namespace of combinedRules) {
+      for (const group of namespace.groups) {
+        for (const rule of group.rules) {
+          const id = ruleId.fromCombinedRule(ruleSourceName, rule);
+
+          if (ruleId.equal(id, identifier)) {
+            return rule;
+          }
+        }
+      }
+    }
+
+    return;
+  }, [identifier, ruleSourceName, combinedRules]);
+
+  return {
+    ...requestState,
+    result: rule,
+  };
+}
+
+export function useCombinedRulesMatching(
+  ruleName: string | undefined,
+  ruleSourceName: string | undefined
+): AsyncRequestState<CombinedRule[]> {
+  const requestState = useCombinedRulesLoader(ruleSourceName);
+  const combinedRules = useCombinedRuleNamespaces(ruleSourceName);
+
+  const rules = useMemo(() => {
+    if (!ruleName || !ruleSourceName || combinedRules.length === 0) {
+      return [];
+    }
+
+    const rules: CombinedRule[] = [];
+
+    for (const namespace of combinedRules) {
+      for (const group of namespace.groups) {
+        for (const rule of group.rules) {
+          if (rule.name === ruleName) {
+            rules.push(rule);
+          }
+        }
+      }
+    }
+
+    return rules;
+  }, [ruleName, ruleSourceName, combinedRules]);
+
+  return {
+    ...requestState,
+    result: rules,
+  };
+}
+
+function useCombinedRulesLoader(ruleSourceName: string | undefined): AsyncRequestState<void> {
+  const dispatch = useDispatch();
+  const promRuleRequests = useUnifiedAlertingSelector((state) => state.promRules);
+  const promRuleRequest = getRequestState(ruleSourceName, promRuleRequests);
+  const rulerRuleRequests = useUnifiedAlertingSelector((state) => state.rulerRules);
+  const rulerRuleRequest = getRequestState(ruleSourceName, rulerRuleRequests);
+
+  useEffect(() => {
+    if (!ruleSourceName) {
+      return;
+    }
+
+    dispatch(fetchPromRulesAction(ruleSourceName));
+    dispatch(fetchRulerRulesAction(ruleSourceName));
+  }, [dispatch, ruleSourceName]);
+
+  return {
+    loading: promRuleRequest.loading || rulerRuleRequest.loading,
+    error: promRuleRequest.error ?? rulerRuleRequest.error,
```

Comment:
```suggestion
    error: promRuleRequest.error ?? isRulerNotSupportedResponse(rulerRuleRequest) ? undefined : rulerRuleRequest.error,
```

This would allow viewing rule from a rules source that does not support ruler. Now it show error with "ruler not supported"